DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces_overview/handshake-control/src/vector_add.cpp

```diff
@@ -0,0 +1,116 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class IDSimpleVAdd;
+
+struct SimpleVAddKernel {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      A_in;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      B_in;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      C_out;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int, decltype(sycl::ext::oneapi::experimental::properties{
+               sycl::ext::intel::experimental::conduit})>
+      len;
+
+  // kernel property method
+  auto get(sycl::ext::oneapi::experimental::properties_tag) {
+    return sycl::ext::oneapi::experimental::properties{
+        sycl::ext::intel::experimental::streaming_interface<>,
+        sycl::ext::intel::experimental::pipelined<1>};
+  }
+
+  void operator()() const {
+    for (int idx = 0; idx < len; idx++) {
+      int a_val = A_in[idx];
+      int b_val = B_in[idx];
+      int sum = a_val + b_val;
+      C_out[idx] = sum;
+    }
+  }
+};
+
+#define VECT_SIZE 256
+
+int main() {
+  try {
+    // Use compile-time macros to select either:
+    //  - the FPGA emulator device (CPU emulation of the FPGA)
+    //  - the FPGA device (a real FPGA)
+    //  - the simulator device
+#if FPGA_SIMULATOR
+    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+    auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler);
+
+    int count = VECT_SIZE;  // pass array size by value
+
+    // declare arrays and fill them
+    // allocate in shared memory so the kernel can see them
+    int *A = sycl::malloc_shared<int>(count, q);
```

Comment:
snake_case for variables here and below
```suggestion
    int *a = sycl::malloc_shared<int>(count, q);
```